src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs

```diff
@@ -100,38 +105,63 @@ public void StartProcessDump(int processId, string dumpFileGuid, string testResu
         /// <returns>Arguments</returns>
         private static string BuildProcDumpArgs(int processId, string filename, bool isFullDump = false)
         {
+            ProcessDumpUtility.procDumpExceptionsList = new List<string>()
+            {
+                "STACK_OVERFLOW",
+                "ACCESS_VIOLATION"
+            };
+
             // -accepteula: Auto accept end-user license agreement
+            // -e: Write a dump when the process encounters an unhandled exception. Include the 1 to create dump on first chance exceptions.
+            // -g: Run as a native debugger in a managed process (no interop).
             // -t: Write a dump when the process terminates.
+            // -ma: Full dump argument.
+            // -f: Filter the exceptions.
+            StringBuilder procDumpArgument = new StringBuilder("-accepteula -e 1 -g -t ");
             if (isFullDump)
             {
-                // This will create a fulldump of the process with specified filename
-                return "-accepteula -t -ma " + processId + " " + filename + ".dmp";
+                procDumpArgument.Append("-ma ");
             }
-            else
+
+            foreach (var exceptionFilter in ProcessDumpUtility.procDumpExceptionsList)
             {
-                // This will create a minidump of the process with specified filename
-                return "-accepteula -t " + processId + " " + filename + ".dmp";
+                procDumpArgument.Append($"-f {exceptionFilter} ");
             }
+
+            procDumpArgument.Append($"{processId} {filename}.dmp");
+            return procDumpArgument.ToString();
```

Comment:
Can we output this argument in logs as well (with info level may be)?

Reply:
Output in logs done